### PR TITLE
Update SubM roadmap docs, describe planning process

### PR DESCRIPTION
**Pull request by dfarrell07**
Make the docs more resilient to not needing updating each release.
Include more information about the planning process.

Closes: #270
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

File: src/content/roadmap/_index.en.md [hunk 1]
```diff
@@ -5,15 +5,24 @@ weight = 40
...
-This is a preliminary community roadmap. It represents our best guess at broad priorities, and can serve as a general guideline.
+Submariner organizes all current and upcoming work using GitHub Issues, Projects, and Milestones.
 
-Details of past releases are described [here](../releases).
+## Planning Process
 
-## v0.7.0
+In preparation for sprint planning meetings (see [SubM's GCal][cal]), GitHub Issues should be raised for work that is to be a part of a
+sprint.  Issues targeted for a sprint should be added to the upcoming Project's "Backlog" column. During sprint planning meetings, Issues
+will be discussed and moved to the "TODO" column and the milestone will be set to the targeted release. As contributors make progress during
```
Comment (nyechiel) [inline src/content/roadmap/_index.en.md:14]:
```suggestion
will be discussed and moved to the "TO DO" column and the milestone will be set to the targeted release. As contributors make progress during
```

File: src/content/roadmap/_index.en.md [hunk 2]
```diff
@@ -5,15 +5,24 @@ weight = 40
...
 +++
 
-This is a preliminary community roadmap. It represents our best guess at broad priorities, and can serve as a general guideline.
+Submariner organizes all current and upcoming work using GitHub Issues, Projects, and Milestones.
 
-Details of past releases are described [here](../releases).
+## Planning Process
 
-## v0.7.0
+In preparation for sprint planning meetings (see [SubM's GCal][cal]), GitHub Issues should be raised for work that is to be a part of a
```
Comment (nyechiel) [inline src/content/roadmap/_index.en.md:12]:
```suggestion
In preparation for sprint planning meetings (see [Submariner's Community Calendar][cal]), GitHub Issues should be raised for work that is to be a part of a
```

**Review by nyechiel — COMMENTED** [review]
Couple of nits, but overall LGTM. Thanks!

File: src/content/roadmap/_index.en.md (hunk 1 above)
Comment (dfarrell07) [reply src/content/roadmap/_index.en.md:14]: The column is "TO DO", but I think it should be "TODO". As I understand, it should either be one word or hyphenated. In code it's typically one word.

https://english.stackexchange.com/questions/46217/todo-list-or-to-do-list

File: src/content/roadmap/_index.en.md (hunk 2 above)
Comment (dfarrell07) [reply src/content/roadmap/_index.en.md]: Done, and re-wrapped. Thanks!

**Review by nyechiel — APPROVED** [review]

**Comment by dfarrell07** [comment]
(removed one extra space)

**Review by tpantelis — APPROVED** [review]

**Pull request merged**